site/content/blog/2023-02-16-nextjs-tutorial.md

```diff
@@ -0,0 +1,218 @@
+
+![[dark-theme.jpg]]
+
+https://lifeitself.org/
+
+![[lifeitself.png]]
+
+https://datahub.io/
+
+![[datahub.png]]
+
+## Creating a new Next.js project using Flowershow template
+
+In order to create a new Next.js project using the Flowershow template, you can use Next.js' `create-next-app` CLI tool.
+
+Run the following command in the terminal and then pass a name of your project when prompted (we'll name it `my-flowershow-app` for the purpose of this tutorial).
+
+```shell
+npx create-next-app@latest --example https://github.com/flowershow/flowershow/tree/main/packages/template
+```
+
+After the command has finished running, your project structure should look similar to this:
+
+```shell
+my-flowershow-app
+├── components
+├── config
+├── layouts
+├── lib
+├── pages
+├── public
+├── scripts
+├── styles
+├── contentlayer.config.ts
+├── next.config.mjs
+├── tailwind.config.cjs
+├── package-lock.json
+├── package.json
+├── ...
+└── README.md
+```
+
+The only thing that's missing is a `/content` folder with some markdown files you'd like to publish.
+
+If you already have a folder with some files, e.g. your blog pages, you can move them to your Flowershow project, like so:
+
+```
+mv <path-to-my-content-folder> my-flowershow-app/content
+```
+
+Your `index.md` file in the root of the `content` folder will be your home page.
+
+> [!NOTE]
+> Check [Next.js documentation](https://nextjs.org/docs/api-reference/create-next-app) to learn more about this command and other options you can run it with.
+
+You should be able to run `npm run dev`, where Contentlayer will fetch all the needed data and transform that data into structured content. Content is structured using a custom schema defined in `contentlayer.config.ts`. Once content is fetched and transformed, it is used to populate various parts of the website , such as text, images, and other media.
+
+### Basic configuration
+
+Flowershow supports setting common configuration values using a `config.js` file, which should be placed in the root of the `/content` folder. With it you can customize some things like website title, navigation links in the navbar, or set your google analytics key.
+
+Let's create one:
+
+```shell
+touch my-flowershow-app/content/config.js
+```
+
+Then add some new configuration to your project eg.
+
+```jsx
+const config = {
+  // title will be displayed on the top of your site
+  title: "My Awesome Blog",
+  // adding a description helps with SEO
+  description: "This is my awesome blog built with Flowershow",
+  // author of site displayed on the bottom of your site
+  author: "John",
+  // logo image
+  authorLogo: "/images/logo.svg",
+  // url to author website
+  authorUrl: "https://john.app/",
+  // links to the pages you want to link to in the navbar
+  navLinks: [{ href: "/about", name: "About" }],
+};
+
+export default config;
+```
+
+> [!NOTE]
+> If you want to learn more about other available configuration options, see https://flowershow.app/docs/config.
+
```

Comment:
I think `npm run dev` will only work here, i.e. after creating the `config.js` file.